### ISSUE TITLE
`InvalidPermission.NotFound` when creating security group rules with overlapping port ranges

### ISSUE THREAD

**Issue by joshuaspence** [terraform-providers/terraform-provider-aws#3261]
With Terraform 0.11.2 and version 1.8.0 of the AWS provider, the following code applies cleanly on the first run, but subsequently fails:

```
resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_security_group" "dummy" {
  vpc_id = "${aws_vpc.main.id}"
  count  = 2
}

resource "aws_security_group" "test" {
  vpc_id = "${aws_vpc.main.id}"

  ingress {
    description     = "A"
    from_port       = 80
    to_port         = 80
    protocol        = "tcp"
    security_groups = ["${aws_security_group.dummy.*.id[0]}"]
  }

  ingress {
    description     = "B"
    from_port       = 80
    to_port         = 80
    protocol        = "tcp"
    security_groups = ["${aws_security_group.dummy.*.id[1]}"]
  }
}
```

After applying the change for the first time, a subsequent plan shows the following changes to be made:

```
  ~ aws_security_group.test
      ingress.1195428013.cidr_blocks.#:              "0" => "0"
      ingress.1195428013.description:                "A" => ""
      ingress.1195428013.from_port:                  "80" => "0"
      ingress.1195428013.ipv6_cidr_blocks.#:         "0" => "0"
      ingress.1195428013.protocol:                   "tcp" => ""
      ingress.1195428013.security_groups.#:          "1" => "0"
      ingress.1195428013.security_groups.2302967246: "sg-731ce90a" => ""
      ingress.1195428013.self:                       "false" => "false"
      ingress.1195428013.to_port:                    "80" => "0"
      ingress.1819122542.cidr_blocks.#:              "0" => "0"
      ingress.1819122542.description:                "" => "B"
      ingress.1819122542.from_port:                  "" => "80"
      ingress.1819122542.ipv6_cidr_blocks.#:         "0" => "0"
      ingress.1819122542.protocol:                   "" => "tcp"
      ingress.1819122542.security_groups.#:          "0" => "1"
      ingress.1819122542.security_groups.2302967246: "" => "sg-731ce90a"
      ingress.1819122542.self:                       "" => "false"
      ingress.1819122542.to_port:                    "" => "80"
      ingress.2787947209.cidr_blocks.#:              "0" => "0"
      ingress.2787947209.description:                "A" => "A"
      ingress.2787947209.from_port:                  "80" => "80"
      ingress.2787947209.ipv6_cidr_blocks.#:         "0" => "0"
      ingress.2787947209.protocol:                   "tcp" => "tcp"
      ingress.2787947209.security_groups.#:          "1" => "1"
      ingress.2787947209.security_groups.1574370408: "sg-e91fea90" => "sg-e91fea90"
      ingress.2787947209.self:                       "false" => "false"
      ingress.2787947209.to_port:                    "80" => "80"
```

Attempting to apply these changes fails:

```
* aws_security_group.test: Error revoking security group ingress rules: InvalidPermission.NotFound: The specified rule does not exist in this security group.
	status code: 400, request id: 31ae42d3-92d5-4d8a-9a89-df10ddfa0d20
```

I believe that the issue here is that the two `ingress` rules have the same port ranges and so the AWS API merges them, which Terraform sees as a change.

**Comment by vaseemkhan007** [comment]
I am also facing the same issue but fixed it for while. I did this like, delete all rule in the SG, run it again. May be issue in the version.
My Version: Terraform v0.11.3

**Comment by bflad** [comment]
Hi folks 👋  Sorry this has been a longstanding issue with the AWS provider. The fix for this should be contained in #4416 which will be released with v1.19.0 of the AWS provider, likely middle of next week.

Shout outs to @loivis (and @svanharmelen who submitted an earlier, likely correct PR, which I admittedly should have reviewed and merged sooner: #3628)

Given there were so many various issues surrounding this bug, I will be locking this issue (amongst all the others) to encourage any lingering issues/discussion to be fully described in new issue(s) for consolidation. Thanks for your understanding.
